scr/main/java/bittercode/util/DBUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.bittercode.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.model.StoreException;
+
+public class DBUtil {
+
+    private static Connection connection;
+
+    static {
+
+        try {
+
+            Class.forName(DatabaseConfig.DRIVER_NAME);
+            
+            connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING, DatabaseConfig.DB_USER_NAME,
+                    DatabaseConfig.DB_PASSWORD);
+        } catch (SQLException | ClassNotFoundException e) {
+
+            e.printStackTrace();
+
+        }

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using connection pooling instead</b></div>
<div id="fix">

Consider moving database connection initialization to a connection pool implementation like HikariCP or C3P0 for better resource management and performance. Static initialization blocks can lead to startup failures that are hard to recover from.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,26 +14,20 @@
 -    static {
 -        try {
 -            Class.forName(DatabaseConfig.DRIVER_NAME);
 -            connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING,
 -                    DatabaseConfig.DB_USER_NAME, DatabaseConfig.DB_PASSWORD);
 -        } catch (SQLException | ClassNotFoundException e) {
 -            e.printStackTrace();
 -        }
 -    }
 +    private static HikariDataSource dataSource;
 +    
 +    static {
 +        HikariConfig config = new HikariConfig();
 +        config.setJdbcUrl(DatabaseConfig.CONNECTION_STRING);
 +        config.setUsername(DatabaseConfig.DB_USER_NAME);
 +        config.setPassword(DatabaseConfig.DB_PASSWORD);
 +        dataSource = new HikariDataSource(config);
 +    }

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2517477681>#c23dbc</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged